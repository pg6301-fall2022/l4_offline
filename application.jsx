import * as React from "react";

export function ListMovies(){
    return <div><h1>List Movies</h1>
        <ul>
            <li>Movie 1</li>
            <li>Movie 2</li>
        </ul>
    </div>;
}

export function NewMovieForm() {
    return <form>
        <h1>Add new movie</h1>
        <div>
            Title: <input/>
        </div>
        <div>
            Year: <input/>
        </div>
        <div>
            Plot:
            <div><textarea/></div>
        </div>
    </form>;
}