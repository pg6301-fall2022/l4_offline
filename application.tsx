import * as React from "react";
import { useState } from "react";

export function ListMovies({ movies }: { movies: string[] }) {
  return (
    <div>
      <h1>List Movies</h1>
      <ul>
        {movies.map((m) => (
          <li key={m}> (m) </li>
        ))}
      </ul>
    </div>
  );
}

interface Movie {
  title: string;
  year: string;
  plot: string;
}

export function NewMovieForm({
  onAddMovie,
}: {
  onAddMovie(movie: Movie): void;
}) {
  const [title, setTitle] = useState("");
  const [year, setYear] = useState("");
  const [plot, setPlot] = useState("");

  function handleSubmit() {
    onAddMovie({ title, year, plot });
  }

  return (
    <form onSubmit={handleSubmit}>
      <h1>Add new movie</h1>
      <div>
        Title:{" "}
        <input
          data-testid="title"
          value={title}
          onChange={(e) => setTitle(e.target.value)}
        />
      </div>
      <div>
        Year:{" "}
        <input
          data-testid="year"
          value={year}
          onChange={(e) => setYear(e.target.value)}
        />
      </div>
      <div>
        Plot:
        <div>
          <textarea
            data-testid="plot"
            value={plot}
            onChange={(e) => setPlot(e.target.value)}
          />
        </div>
      </div>
    </form>
  );
}
