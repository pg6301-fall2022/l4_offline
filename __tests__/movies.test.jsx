import * as React from "react";
import * as ReactDOM from "react-dom";

import {ListMovies, NewMovieForm} from "../application";

describe ("movies application", () => {
    it("shows movie list", () => {
        const element = document.createElement("div");
        ReactDOM.render(<ListMovies/>, element);
        expect(element.querySelector("h1").innerHTML)
            .toEqual("List Movies");
        expect(element.innerHTML).toMatchSnapshot();
    });

    it("shows new movie form", () => {
        const element = document.createElement("div");
        ReactDOM.render(<NewMovieForm/>, element);
        expect(element.innerHTML).toMatchSnapshot();
    })

    })