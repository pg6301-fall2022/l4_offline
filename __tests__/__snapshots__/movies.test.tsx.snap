// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movies application shows movie list 1`] = `"<div><h1>List Movies</h1><ul><li> (m) </li><li> (m) </li></ul></div>"`;

exports[`movies application shows new movie form 1`] = `""`;
